lib/open_api_spex/cast/enum.ex

```diff
@@ -0,0 +1,48 @@
+defmodule OpenApiSpex.Cast.Enum do
+  @moduledoc false
+  alias OpenApiSpex.Cast
+
+  def cast(%Cast{schema: %{enum: []}} = ctx) do
+    Cast.error(ctx, {:invalid_enum})
+  end
+
+  def cast(%Cast{schema: %{enum: [value | _]}, value: value}) do
+    {:ok, value}
+  end
+
+  # Special case: convert binary to atom enum
+  def cast(ctx = %Cast{schema: schema = %{enum: [atom_value | tail]}, value: value})
+      when is_binary(value) and is_atom(atom_value) do
+    if value == to_string(atom_value) do
+      {:ok, atom_value}
+    else
+      cast(%{ctx | schema: %{schema | enum: tail}})
```

Comment:
We've been trying to avoid modifying the schema during casting. It makes reasoning and debugging more difficult. But if avoiding it has too many of its own drawbacks, then that's fine.

Reply:
👍